src/precice/SolverInterface.hpp

```diff
@@ -772,6 +772,47 @@ class SolverInterface {
       int     valueIndex,
       double &value) const;
 
+  /**
+   * @brief setBoundingBox Define a region of interest in order to filter a
+   *        received mesh for a certain mesh region
+   *
+   * @param[in] dataID ID of the Data you want to access through the bounding box
+   * @param[in] boundingBox (axis aligned) which has in 3D the format
+   *            [x_min, x_max, y_min, y_max, z_min, z_max]
+   *
+   * @return ID of the bounding box (similar to a mesh ID)
+   */
+  int setBoundingBox(
+      const int     dataID,
+      const double *boundingBox) const;
+
+  /**
+   * @brief getLocallyRelevantMeshSize
+   *
+   * @param[in] dataID of the associated mesh. Most probably the data ID of your local readData
+   *
+   * @return Size of the locally relevant mesh as given due to the bounding box
+   *
+   * @pre bounding box has been defined using @p setBoundingBox
+   */
+  int getLocallyRelevantMeshSize(const int dataID) const;
+
+  /**
+   * @brief getLocallyRelevantCoordinates Iterates over the region of
+   *        interest defined by bounding boxes and reads the corresponding
+   *        coordinates omitting the mapping.
+   *
+   * @param[in]  dataID corresponding data ID
+   * @param[out] coordinates associated to the values (dim * @p getLocallyRelevantMeshSize)
+   *
+   * @pre Values and coordinates have the correct size, which can be requested by getLocallyRelevantMeshSize)
+   * @pre bounding box has been defined using @p setBoundingBox
+   */
+  void getLocallyRelevantCoordinates(
```

Comment:
This is exactly `getMeshVertices`, right? I don't think we need an additional API function or am I missing sth?

Reply:
This is probably the biggest point to discuss: the problem is that `getMeshVertices` has the corresponding vertex IDs as an input argument. They are, however, unknown to the participant since he has not set the vertices. I guess we need something different here than `getMeshVertices`